docs/cpu-metrics-in-fleet.md

```diff
@@ -0,0 +1,380 @@
+      (essentially dividing the 1-minute bucket value by 6). This is all done by the `cpu_derivative` aggregation in the
+      Elasticsearch query.
+   5. this per-10-second CPU utilization (in milliseconds) is divided by `10 * 1000` (= 10 seconds expressed as milliseconds),
+      to get the _percentage_ CPU utilization over 10 elapsed seconds. This is the `cpu` aggregation in the Elasticsearch query.
+   6. So now we end up with five CPU utilization (in %) values, one for each minute.
+   7. These are averaged to arrive at the CPU utilization (in %) over five minutes. This is the `avg_cpu` aggregation in
+      the Elasticsearch query.
+2. The 5-minute average CPU utilization (in %) for each of the processes is summed up to arrive at a total CPU utilization
+   (in %) for the Agent process and all Beat processes managed by Agent. This is the `sum_cpu` aggregation in the Elasticsearch
+   query. The result is the total CPU utilization (in %) by Agent and it's child Beat processes across all available cores
+   (as opposed to being normalized per-core). For example, if a machine has 8 cores, the resulting value will be in the range of
+   (0%, 800%).
+
+## Observations
+
+* The CPU utilization (in %) of Agent and each of the Beat processes can vary wildly. Generally speaking, the Agent process
+  itself does not utilize much CPU. And each Beat process may utilize CPU depending on the type of computations it is performing
+  on the data.
+
+* Also, CPU utilization is rarely constant. If the output of `top` or `htop` for Agent and Beat processes is observed over
+  time, the CPU utilization % shown varies for each process over time.
+
+* To relate the output seen in `top` or `htop` for Agent and Beat processes with the single value shown in the Fleet UI,
+  one must observe the values in the `top` / `htop` output over five minutes, arrive at an average CPU utilization (in %)
+  value for each process, and sum it up.  The resulting value will roughly be equal to the single value shown for that Agent
+  in the Agent Listing page in the Fleet UI.
+
+## Suggested improvements
+
+* The `processes` aggregation in the Elasticsearch query should use the field `component.id` instead of `elastic_agent.process`.
+  This is to correctly account for multiple instances of the same type of Beat (e.g. Filebeat).  This can happen if there
+  are multiple outputs defined in the Agent policy and some inputs of a type (e.g. log) use one output while other inputs
+  of the same type use another output.
+
+* We should reconsider taking a 5-minute average in the Elasticsearch query made by the Fleet UI and instead take a
+  30-second or 1-minute average (making corresponding adjustments to the `calendar_interval` value in the `cpu_time_series`
+  aggregation). This would result in a value closer to what's observed in `top` / `htop` output.
+
+* We should link the value shown in the Fleet UI to a chart that breaks it down for that Agent by `component.id` over time,
```

Comment:
should this be the `elastic_agent` dashboard?

Reply:
Yes, specifically the Agent Metrics one, e.g.:

<img width="1588" alt="Screenshot 2024-01-11 at 11 51 54" src="https://github.com/elastic/elastic-agent/assets/51061/fb8d74f8-83c4-4990-895c-3fdce796c301">